### ISSUE TITLE
RFC: do not use `_serverProcess` in leanclient.

### ISSUE THREAD

**Issue by joneugster** [leanprover/vscode-lean4#487]
Consider the following lines of code: 

https://github.com/leanprover/vscode-lean4/blob/cee53169b58d31a03a60c4769ee827a3cabffd7c/vscode-lean4/src/leanclient.ts#L264C1-L274C11

```
;(this.client as any)._serverProcess.stderr.on('data', async (chunk: Buffer) => {
  ...
})
```

When adapting the extension to monaco, we are replacing vscode's `LanguageClient` ([link to source](https://github.com/microsoft/vscode-languageserver-node/blob/33cbe5c87032cb53cf30c2b90f29d22a45ee6800/client/src/node/main.ts#L128-L134)) with the `MonacoLanguageClient` ([link to source](https://github.com/TypeFox/monaco-languageclient/blob/585530dfee5a2929c45dddc4ebc930351cb36ba0/packages/client/src/index.ts#L19-L30)) which does not have `_serverProcess`, so we basically delete the marked code snippet.

As `(this.client as any)._serverProcess` seems like quite a hack already (i.e. `_serverProcess` is marked `private`), I thought I'd mark this here. Maybe there is a better way to do this that would work in both cases?

**Comment by abentkamp** [comment]
I don't think this is easy to change since the Lean server would need to be modified for this, too.

How about just adding a few question marks?
```
;(this.client as any)._serverProcess?.stderr?.on('data', async (chunk: Buffer) => {
  ...
})
```
